File: 250708/그래프 탐색/graph-traversal.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const edges = [];
for (let i = 1; i <= m; i++) {
    edges.push(input[i].split(' ').map(Number));
}

const graph = Array.from({length: n + 1}, () => []);
for(const [a, b] of edges){
    graph[a].push(b);
    graph[b].push(a);
}

const visited = Array(n + 1).fill(false);

function dfs(node){
    visited[node] = true
    for(const next of graph[node]){
        if(!visited[next]){
            dfs(next);
        }
    }
}
dfs(1)

console.log(visited.reduce((prev, cur) => prev + (cur ? 1 : 0), 0) - 1)

File: 250820/그래프 탐색/graph-traversal.py
n, m = map(int, input().split())
edges = [tuple(map(int, input().split())) for _ in range(m)]

# Please write your code here.
graph = [[] for _ in range(n+1)]

visited = [False for _ in range(n+1)]

count = 0

for v1, v2 in edges:
    graph[v1].append(v2)
    graph[v2].append(v1)

def dfs(v):
    global count

    for cur in graph[v]:
        if not visited[cur]:
            visited[cur] = True
            count += 1
            dfs(cur)

visited[1] = True
dfs(1)

print(count)